### PR TITLE
#184948277 Review a Product

### PR DESCRIPTION
**Pull request by ericndungutse**
#### What does this PR do?
This PR implements **Review** feature

#### Description of Task to be completed?

- [x] Create Review form on product detail page
- [x] Render reviews of a product on product detail page
- [x]  Integrate with backend

#### How should this be manually tested?
- In your project terminal, run `git fetch origin pull/40/head:ft-review-products-184948277`
- Checkout to **ft-signup-with-email-and-password-184948250** branch
- Install all the dependencies by running npm install
- create `.env` file and set `LIVE_API_URL` to server api `url`
- Then run the project by typing `npm run dev`

#### Routes to be tested:
- `product/<product-id>`

#### What are the relevant pivotal tracker/Trello stories?
Story ID: [184948277](https://www.pivotaltracker.com/story/show/184948277)

#### Screenshoots

Screenshot (25)
Screenshot (26)

### CODE REVIEW

**Review by GasanaJr — APPROVED** [review]
LGTM

**Review by Rwibutso — APPROVED** [review]

File: src/components/review/Reviews.jsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+									(hoverRate >= 2 || rate >= 2) &&
+									`rating-star--active`
+								} `}
+								onMouseEnter={(e) =>
+									handleOnMouseEnter(e, setHoverRate)
+								}
+								onMouseLeave={() =>
+									handleOnMouseLeave(setHoverRate)
+								}
+								onClick={(e) => handleOnClick(e, setRate)}
```
Comment (Habinezajanvier) [inline src/components/review/Reviews.jsx]: Can't reduce lines of codes by putting those Stars in a Array.map()?

File: src/components/review/review.utils.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		const response = await fetch(
+			`${process.env.BASE_URL}/reviews/${id}/review`,
+			{
+				method: "POST",
+				headers: reqHeaders,
+				body: JSON.stringify(feedback),
+			},
+		);
+		
+		const data = await response.json();
```
Comment (Habinezajanvier) [inline src/components/review/review.utils.js]: Can we put all request handles in redux actions?

**Review by Habinezajanvier — DISMISSED** [review]

**Comment by Emile-SHUMBUSHO** [comment]
LGTM and I would like to suggest, reduce the size of that textArea and justify this content to flex-start instead of being centered.
Screenshot 2023-05-18 at 17 46 49

**Review by wilbrord2 — APPROVED** [review]

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**